source/distributed/fully_distributed_tria.cc

```diff
@@ -285,6 +289,71 @@ namespace parallel
 
 
 
+    template <int dim, int spacedim>
+    void
+    Triangulation<dim, spacedim>::copy_triangulation(
+      const dealii::Triangulation<dim, spacedim> &other_tria)
+    {
+      Assert(this->dynamic_construction_data,
+             ExcMessage("No DynamicConstructionData has been provided!"));
+
+      // pointer to the triangulation for which the construction data
+      // should be created (normally it is the input triangulation but
+      // in the case of a serial triangulation we create a copy which should
+      // be used)
+      const dealii::Triangulation<dim, spacedim> *other_tria_ptr = &other_tria;
+
+      // temporary serial triangulation (since we might modifying subdomain_ids
+      // and level_subdomain_ids during partitioning); it is filled only
+      // if the source triangulation is serial
+      dealii::Triangulation<dim, spacedim> serial_tria;
+
+      // check if other triangulation is not a parallel one, which needs to be
+      // partitioned
+      if (dynamic_cast<const dealii::parallel::TriangulationBase<dim, spacedim>
+                         *>(&other_tria) == nullptr)
+        {
+          // actually copy the serial triangulation
+          serial_tria.copy_triangulation(
```

Comment:
@kronbichler  Is this more after your liking?

Reply:
Yes, I think this is much simpler. My only question was whether there was a cost/memory consumption reason - but I guess we can tolerate that here.